File: c_plus_plus/tests/singleroundtests/gamestagestest.cpp
#include "gamestagestest.h"
#include <QTest>
#include "abstractplaneround.h"



void GameStagesTest::initTestCase()
{
    qDebug("GameStagesTest starts ..");
}

void GameStagesTest::GameStages_Values() {
    QVERIFY(0 == (int)AbstractPlaneRound::GameStages::GameNotStarted);
    QVERIFY(1 == (int)AbstractPlaneRound::GameStages::BoardEditing);
    QVERIFY(2 == (int)AbstractPlaneRound::GameStages::Game);
    QVERIFY(3 == (int)AbstractPlaneRound::GameStages::WaitForOpponentPlanesPositions);
}


void GameStagesTest::cleanupTestCase()
{
    qDebug("GameStagesTest ends ..");
}

File: kotlin/PlanesAndroid/app/src/test/java/com/planes/singleround/test/GameStagesTest.kt
package com.planes.singleround.test.com.planes.singleround.test
import com.google.common.truth.Truth
import com.planes.single_player_engine.GameStages
import org.junit.Test

class GameStagesTest {
    @Test
    fun GameStages_Values() {
        Truth.assertThat(GameStages.values()[0] === GameStages.GameNotStarted).isTrue()
        Truth.assertThat(GameStages.values()[1] === GameStages.BoardEditing).isTrue()
        Truth.assertThat(GameStages.values()[2] === GameStages.Game).isTrue()
        Truth.assertThat(GameStages.values()[3] === GameStages.WaitForOpponentPlanesPositions).isTrue()
        Truth.assertThat(GameStages.values()[4] === GameStages.WaitForOpponentMoves).isTrue()
        Truth.assertThat(GameStages.values()[5] === GameStages.SendRemainingMoves).isTrue()
    }
}

